easybuild/easyblocks/s/siesta.py

```diff
@@ -0,0 +1,318 @@
+
+@author: Miguel Dias Costa (National University of Singapore)
+@author: Ake Sandgren (Umea University)
+"""
+import os
+import stat
+
+import easybuild.tools.toolchain as toolchain
+from distutils.version import LooseVersion
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import adjust_permissions, apply_regex_substitutions, change_dir, copy_dir, copy_file, mkdir
+from easybuild.tools.modules import get_software_root
+from easybuild.tools.run import run_cmd
+
+
+class EB_Siesta(ConfigureMake):
+    """
+    Support for building/installing Siesta.
+    - avoid parallel build for older versions
+    """
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Define extra options for Siesta"""
+        extra = {
+            'with_transiesta': [True, "Build transiesta", CUSTOM],
+            'with_utils': [True, "Build all utils", CUSTOM],
+        }
+        return ConfigureMake.extra_options(extra_vars=extra)
+
+    def configure_step(self):
+        """
+        Custom configure and build procedure for Siesta.
+        - There are two main builds to do, siesta and transiesta
+        - In addition there are multiple support tools to build
+        """
+
+        start_dir = self.cfg['start_dir']
+        obj_dir = os.path.join(start_dir, 'Obj')
+        arch_make = os.path.join(obj_dir, 'arch.make')
+        bindir = os.path.join(self.cfg['start_dir'], 'bin')
+
+        netcdff_loc = get_software_root('NetCDF-Fortran')
+
+        par = ''
+        if LooseVersion(self.version) >= LooseVersion("4.1"):
+            par = '-j %s' % self.cfg['parallel']
+
+        # enable OpenMP support if desired
+        env_var_suff = ''
+        openmp = self.toolchain.options.get('openmp', None)
+        if openmp:
+            env_var_suff = '_MT'
+
+        scalapack = os.environ['LIBSCALAPACK' + env_var_suff]
+        blacs = os.environ['LIBSCALAPACK' + env_var_suff]
+        lapack = os.environ['LIBLAPACK' + env_var_suff]
+        blas = os.environ['LIBBLAS' + env_var_suff]
+        if get_software_root('imkl') or get_software_root('FFTW'):
+            fftw = os.environ['LIBFFT' + env_var_suff]
+
+        regex_subs = [
+            ('dc_lapack.a', ''),
+            (r'^NETCDF_INTERFACE\s*=.*$', ''),
+            ('libsiestaBLAS.a', ''),
+            ('libsiestaLAPACK.a', ''),
+            # Needed here to allow 4.1-b1 to be built with openmp
+            (r"^(LDFLAGS\s*=).*$", r"\1 %s %s" % (os.environ['FCFLAGS'], os.environ['LDFLAGS'])),
+        ]
+
+        if netcdff_loc:
+            # Needed for gfortran at least
+            regex_subs.append((r"^(ARFLAGS_EXTRA\s*=.*)$", r"\1\nNETCDF_INCFLAGS = -I%s/include" % netcdff_loc))
+
+        if fftw:
+            fftw_inc, fftw_lib = os.environ['FFTW_INC_DIR'], os.environ['FFTW_LIB_DIR']
```

Comment:
You should be using `FFT_INC_DIR` and `FFT_LIB_DIR` here, which are also defined if `imkl` is being used

This was fixed in https://github.com/akesandgren/easybuild-easyblocks/pull/4, together with a bunch of minor style fixes.